cabal-install/Distribution/Client/CmdRun.hs

```diff
@@ -326,11 +328,13 @@ selectPackageTargets targetSelector targets
 selectComponentTarget :: PackageId -> ComponentName -> SubComponentTarget
                       -> AvailableTarget k -> Either TargetProblem  k
 selectComponentTarget pkgid cname subtarget@WholeComponent t
-  | CExeName _ <- availableTargetComponentName t
-  = either (Left . TargetProblemCommon) return $
-           selectComponentTargetBasic pkgid cname subtarget t
-  | otherwise
-  = Left (TargetProblemComponentNotExe pkgid cname)
+  = case availableTargetComponentName t
```

Comment:
amend the haddock of `selectComponentTarget`.

We should be precise here when we mean "executable" and when "executable-like" component. This will cause confusion otherwise.